<!DOCTYPE html>
<html lang="en" data-bs-theme="light">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>SNP Query - RiceSNP-Seek Complete Help Manual</title>
        <link href="../css/bootstrap.min.css" rel="stylesheet">
        <link href="../css/fontawesome.min.css" rel="stylesheet">
        <link href="../css/brands.min.css" rel="stylesheet">
        <link href="../css/solid.min.css" rel="stylesheet">
        <link href="../css/v4-font-face.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link id="hljs-light" rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" >
        <link id="hljs-dark" rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github-dark.min.css" disabled>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
        <script>hljs.highlightAll();</script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-primary">
            <div class="container">
                <a class="navbar-brand" href="..">RiceSNP-Seek Complete Help Manual</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-bs-toggle="collapse" data-bs-target="#navbar-collapse" aria-controls="navbar-collapse" aria-expanded="false" aria-label="Toggle navigation">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="nav-item">
                                <a href="../Help/" class="nav-link">Help</a>
                            </li>
                            <li class="nav-item">
                                <a href="./" class="nav-link active" aria-current="page">SNP Query</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Search%20result%20views/" class="nav-link">Search result views</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Steps%20in%20SNP%20Query/" class="nav-link">Steps in SNP Query</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Tree%20Browser/" class="nav-link">Tree Browser</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Use%20cases/" class="nav-link">Use cases</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Variety%20Query/" class="nav-link">Variety Query</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Web%20Interface/" class="nav-link">Web Interface</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Locus%20Query/" class="nav-link">Locus Query</a>
                            </li>
                            <li class="nav-item">
                                <a href="../MDS%20Plot/" class="nav-link">MDS Plot</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Multiple%20reference/" class="nav-link">Multiple reference</a>
                            </li>
                            <li class="nav-item">
                                <a href="../My%20Lists/" class="nav-link">My Lists</a>
                            </li>
                            <li class="nav-item">
                                <a href="../New%20topic/" class="nav-link">New topic</a>
                            </li>
                            <li class="nav-item">
                                <a href="../RiceSNP-SeekDatabase%20-%20New/" class="nav-link">RiceSNP-SeekDatabase - New</a>
                            </li>
                            <li class="nav-item">
                                <a href="../RiceSNP-SeekDatabase/" class="nav-link">RiceSNP-SeekDatabase</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Search%20by%20Country_%20Subpopulation/" class="nav-link">Search by Country  Subpopulation</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Search%20by%20Passport_%20Phenotype/" class="nav-link">Search by Passport  Phenotype</a>
                            </li>
                            <li class="nav-item">
                                <a href="../All%20Varieties%20result/" class="nav-link">All Varieties result</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Background/" class="nav-link">Background</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Data%20Generation/" class="nav-link">Data Generation</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Download/" class="nav-link">Download</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Gene%20Locus%20Naming/" class="nav-link">Gene Locus Naming</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Genome%20Alignments/" class="nav-link">Genome Alignments</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Genome%20Browser/" class="nav-link">Genome Browser</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Get%20Variety%20Information/" class="nav-link">Get Variety Information</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Help/" class="nav-link">Help</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Introduction/" class="nav-link">Introduction</a>
                            </li>
                            <li class="nav-item">
                                <a href="../2-Varieties%20result/" class="nav-link">2-Varieties result</a>
                            </li>
                            <li class="nav-item">
                                <a href="../About/" class="nav-link">About</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Advance%20SNP%20Query/" class="nav-link">Advance SNP Query</a>
                            </li>
                            <li class="nav-item">
                                <a href="../All%20Varieties%20result/" class="nav-link">All Varieties result</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Background/" class="nav-link">Background</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Data%20Generation/" class="nav-link">Data Generation</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Download/" class="nav-link">Download</a>
                            </li>
                            <li class="nav-item">
                                <a href="../Gene%20Locus%20Naming/" class="nav-link">Gene Locus Naming</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ms-md-auto">
                        <li class="nav-item">
                            <a href="#" class="nav-link" data-bs-toggle="modal" data-bs-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li class="nav-item">
                                <a rel="prev" href="../Help/" class="nav-link">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li class="nav-item">
                                <a rel="next" href="../Search%20result%20views/" class="nav-link">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-bs-toggle="collapse" data-bs-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-body-tertiary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-bs-level="1"><a href="#snp-query" class="nav-link">SNP Query</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            
            <li class="nav-item" data-bs-level="1"><a href="#_1" class="nav-link"></a>
              <ul class="nav flex-column">
              </ul>
            </li>
            
            <li class="nav-item" data-bs-level="1"><a href="#steps-to-query-snps" class="nav-link">Steps to query SNPs</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            
            <li class="nav-item" data-bs-level="1"><a href="#_2" class="nav-link"></a>
              <ul class="nav flex-column">
              </ul>
            </li>
            
            <li class="nav-item" data-bs-level="1"><a href="#_3" class="nav-link"></a>
              <ul class="nav flex-column">
              </ul>
            </li>
            
            <li class="nav-item" data-bs-level="1"><a href="#_4" class="nav-link"></a>
              <ul class="nav flex-column">
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="snp-query">SNP Query</h1>
<p><img alt="" src="../lib/figure_5.png" /></p>
<p>The SNP Query interface has two functionalities, to compare alleles between two specified varieties, and to compare multiple varieties with a reference genome. In both either case, the query region can be specified either by chromosomal region, by gene locus, or by a list of positions.</p>
<h1 id="_1"></h1>
<h1 id="steps-to-query-snps">Steps to query SNPs</h1>
<ol>
<li>Specifiy varieties</li>
<li>To compare alleles between two varieties - in the two pairwise boxes, type or select from the drop-down list the varieties to compare. The text boxes have auto-completion feature so you only need to type the first few characters then select from the drop-down list. The choice items and recognized terms can be the variety name or accession based on the selected button at the right.</li>
<li>To compare the reference with a subpopulation - select the subpopulation from the listbox. Some items in a choices is a group of more than one subpopulation, all indica (ind1,ind2,ind3,indx), all japonica (temp,trop,temp/trop,trop/temp) and  all varieties </li>
<li>
<p>To compare reference with a list of varieties - select the variety list name in My Variety List, which are defined in My Lists.</p>
</li>
<li>
<p>Specify genome region</p>
</li>
<li>Select the Reference Genome - select among the five sequenced rice genomes to use as reference genome (Japonica Nipponbare, 93-11, IR64-21, DJ123, Kasalath). The chromosome/scaffold and gene locus selections depends on the selected reference genome. </li>
<li>Check Show all reference alleles – to display the alleles of all 5 reference genomes for each SNP position. The alleles and position conversions between reference genomes are based on pairwise alignment results.</li>
<li>To sepecify Chromosome/Scaffold region - select the chromosome/scaffold name, type the start and end base positions. Chromosome is avaiable when a fully assembled genome (Nipponbare, Kasalath) is selected, otherwise (IR64-21, DJ123) scaffold is used. 93-11 has both chromosomes and scaffolds.</li>
<li>To specify Gene locus - type the Gene locus name in the text box. For Nipponbare, we are using the MSUv7 (LOC_OSXXgYYYYYY), RAP (OsXXgYYYYYY), and our merged MSUv71,RAP,FGenesh++ (OsNippoXXgYYYYYY) gene loci names described in NAR Supplementary Materials.  For the other 4 reference genomes, we are proposing a new locus naming convention also described below. The gene text box also has auto-completion feature. When a locus is selected, the chromosomal position boxes are automatically filled with the selected gene's region. </li>
<li>To use a list of SNP positions, select the SNP List previously defined.</li>
<li>To use a list of gene loci, select the Locus List previously defined.</li>
<li>
<p>A genotype can be defined from a SNP List with allele value for each position. You can Match Genotype between this list and all varieties in the selected dataset. The resulting varieties are ranked by the number of matches with the SNP List. Instead of providing a SNP Liist, the genotype could be that of the selected Variety at the specified region.  </p>
</li>
<li>
<p>Set options</p>
</li>
<li>Include indels - include indels in the query</li>
<li>Show phenotype - include the values of the selected phenotype in the results table </li>
<li>Dataset - select the dataset to query (3kRGP, HDRA, both). If both is selected, there is addition option to use either the union or intersection of the SNP positions.</li>
<li>SNP coloring determines the color of alleles as displayed in the table and in the Genome Browser. Options are:</li>
<li>Reference mismatch – SNPs which have polymorphism with the reference are colored red, while the rest is white.</li>
<li>
<p>Nucleotide – SNPs are colored based on the base nucleotide: A green, T red, G orange, C blue. This follows the JBrowse color code.</p>
</li>
<li>
<p>3k SNP Set - select the subset of the 3kRGP SNPs to query (All 32M, Base 18M, Filtered 4.8M, Core 400k). The subsetting criteria is described below.</p>
</li>
<li>Mismatch Only -- if checked, only SNPs having allele mismatches are fetched in the pairwise comparison, or having mismatch with the reference in the multiple comparison.</li>
<li>Include SNPs -- select to include/exclude/highlight Non-synonymous SNPs.</li>
<li>
<p>Missing allele - specify how missing allele is to be counted (ignore means 0 match, 0 mismatch; or 0.5 match 0.5 mismatch)</p>
</li>
<li>
<p>Search - to perform query and display the result in a Table, or view in JBrowse or a Tree</p>
</li>
</ol>
<p>For queries above 50kb region, or 5000 SNPs, or 10 gene loci, the results are not displayed but available for download only.</p>
<p>Maximum allowed queries are 5Mb region, or 500k SNPs, or 1000 gene loci.</p>
<ol>
<li>Reset -  clear all input fields.</li>
</ol>
<h1 id="_2"></h1>
<p>All Varieties result views</p>
<p>Several views are available for SNP query results</p>
<p>Table  - this view displays the Alleles for each variety in a row, for all SNP positions within  the specified region or gene. The color is based on SNP coloring selected. The varieties are sorted by decreasing number of allele mismatches by default. The resulting table can be filtered using any of the column values, and sorted based on any column value by clicking  the column header.</p>
<p>To download the result, click the CSV, Tab, Plink or Flapjack button to get the given format.</p>
<p><img alt="" src="../lib/NewItem54.png" /></p>
<p>At the bottom of the table is the plot showing the allele/genotype frequencies/counts for all SNP positions in the queried region.</p>
<p><img alt="" src="../lib/NewItem55.png" /></p>
<p>JBrowse – the JBrowse genome browser lays out the SNP genotyping result in the SNP Genotyping track. Each variety is represented as a thin row in the track, and sorted in the same order as in the table. This track is dynamically generated from the query result, and covers only the queried region or locus. Zooming out from the initial region gives empty rows at both sides. To view a wider region, do a query again specifying the new wider region. Unlike in the Table view where results are displayed in several pages, Jbrowse displays all varieties having allele mismatch in the queried region at once.</p>
<p><img alt="" src="../lib/NewItem7.png" /></p>
<p>PhyloTree -- Using the number of allele mismatches between varieties within the queried region or gene to compute for distance, the phylogenetic tree is constructed by Neighbor-Joining alorithm. Note that this operation takes some time (~5 minutes for a typical gene length).</p>
<p>Compare genomes -- Displays the alignment of the selected reference and region with the 4 other reference genomes in Vista.</p>
<p><img alt="" src="../lib/NewItem53.png" /></p>
<p>Alignment – view the results in the familiar multiple sequence alignment format. But not that the positions here are not contiguous. It uses the JSAV library.</p>
<p><img alt="" src="../lib/NewItem56.png" /></p>
<h1 id="_3"></h1>
<p>Querying Indels - include INDELs in the query, check the Include Indels checkbox. The indels are displayed in multiple sequence alignment format along with te SNPs. For insertions, the insertion string is inserted in each variety while the reference sequence is shifted right with gaps (-) equal to the longest insertion. For deletion, the reference alleles are inserted to the reference of length equal to the longest deletion, while gaps (-) are inserted to the varieties. The positions are colored green and blue for insertion and deletion regions, respectively.</p>
<p><img alt="" src="../lib/NewItem40.png" /></p>
<p><img alt="" src="../lib/NewItem33.png" /></p>
<p>Multiple Reference Genome</p>
<ol>
<li>
<p>Select the reference genome to use as coordinate (Japonica Nipponbare, IR64-21, DJ123, 93-11, Kasalath)</p>
</li>
<li>
<p>Check Show all reference alleles - display the alleles in the queried region of the 5 reference genomes</p>
</li>
</ol>
<p><img alt="" src="../lib/NewItem30.png" /></p>
<h1 id="_4"></h1>
<p>Two varieties result views</p>
<p>For two varieties query, results can be viewed in a table or in JBrowse.</p>
<p>Table - when two varieties are compared, the result is a list of SNP position within the specified region. The allele of the reference, and the two varieties for each position are displayed in a row. Polymorphism between the two varieties is highlighted with color red.</p>
<p><img alt="" src="../lib/NewItem6.png" /></p>
<p>JBrowse – the JBrowse tab displays the SNPs of the two varieties within the genome, along with the genes. The boxes are colored based on the selected SNP coloring option.</p>
<p><img alt="" src="../lib/NewItem5.png" /></p>
<p>from NAR 2016, Supplementary Materials</p>
<p>Nipponbare merged gene models and names</p>
<p>Four major annotations for Nipponbare – MSU v7 (16), RAP-representative, RAP-predicted (17) and Fgenesh++ (18) – were added as tracks in JBrowse (https://snp-seek.irri.org/_jbrowse.zul). To facilitate curating results, we merged these four annotations into one set of loci. Having a single set of loci simplifies query and analysis of genome and variant features, like non-synonymous SNPs. The strategy for merging gene loci was:</p>
<ol>
<li>Collect all gene (locus) features from all 4 annotations.</li>
<li>Sort all loci by chromosome and by start and end positions.</li>
<li>Determine loci that have at least 50% overlap for the coding region (CDS) and merge those from the same strand into a common, new locus.</li>
<li>The merged locus is named using the convention described in Supplement II with the form OsNippo{YY}g{NNNNNN}, where YY is the chromosome number (01-12), and NNNNNN is the locus count for the chromosome, starting from 10050, with increment of 50. Each locus is also referenced back to their original ID using the Dbxref attribute. Synonyms and Ontology terms (GO, PO, TO) collected from Oryzabase (21) are also added to the created locus in the Alias and Ontology term attributes. These synonyms and ontology attributes are loaded in the CHADO database. The gene function from MSU7 is used for the merged locus, unless it is ‘hypothetical’, ‘unknown’, or ‘expressed’.  For these cases, the functional annotation from RAP is used.</li>
</ol>
<p>Reference genomes gene locus names</p>
<p>We used the gene models published with each of the genomes. These models use the name originally provided by their respective gene prediction/curation pipelines. To avoid displaying these cryptic names in the viewers and queries, we define a naming convention for rice gene models for any newly sequenced reference genome. </p>
<ol>
<li>Collect all gene (locus) features from the original source.</li>
<li>Sort all loci by chromosome and start and end positions.</li>
<li>Gene names start with “Os” for Oryza sativa, and the next five characters define the variety: “Nippo” (Nipponbare), “9311_” (93-11), “Kasal” (Kasalath), “DJ123” (DJ 123) and “IR64_” (IR 64).</li>
<li>Assign chromosome or contig number:</li>
<li>If there is chromosome assembly, the next 2 digits are the chromosome number (“01”-“12”).</li>
<li>
<p>If there is no chromosome assembly, then the next 5 digits are the contig number. The contig number is a rank given to the contigs sorted by decreasing length, where the longest is assigned “00001”. If contigs are of the same length, they are sorted by the sequence in alphabetical order. </p>
</li>
<li>
<p>The next character is “g”, which stands for genomic DNA.  Instead of “g”, it can be “m” for mitochondrial or “c” for chloroplast DNA.</p>
</li>
<li>With genes sorted by start and then by end positions, the next 6 digits are the gene number in the chromosome or contig with an increment of 50. The first gene is assigned number “010050”, the second “010100”, the third “010150”, etc. Gene names in the chromosomes will be preserved in future versions of annotations. Gene names in scaffolds/contigs are for temporary use.</li>
<li>Each locus is also referenced back to their original ID as provided by the source in the Dbxref attribute. If available, synonyms and Ontology terms (GO, PO, TO) are also added to the created locus in the Alias and Ontology term attributes.</li>
<li>Alternatively spliced gene models or different versions of the same gene prediction are distinguished by a number added at the end of the gene after the dot. Here are some examples: “OsNippo01g210150.1”, “OsIR64_00342g210150.1”, “Os9311_11g210150.1”, “OsDJ12_00342g210150.1”.</li>
<li>Each mRNA has its own list of features (CDS, exon, 3’-UTR and 5’-UTR) collected from the original annotation source, with descriptors following the transcript name.  The transcript name is followed by “:” (colon), then the feature type using Sequence Ontology terms (“cds”, “exon”, “five_prime_UTR”, “three_pime_UTR”), a “.” (dot), and the last two digits are the order number. Examples are:  “OsNippo01g210150.1:cds.01” and “Os9311_11g210150.2:exon.01”.</li>
</ol>
<p>Created with the Personal Edition of HelpNDoc: <a href="http://www.helpndoc.com/help-authoring-tool">Free help authoring environment</a></p></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script src="../js/bootstrap.bundle.min.js"></script>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js"></script>
        <script src="../search/main.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="searchModalLabel" aria-hidden="true">
    <div class="modal-dialog modal-lg">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="searchModalLabel">Search</h4>
                <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
            </div>
            <div class="modal-body">
                <p>From here you can search these documents. Enter your search terms below.</p>
                <form>
                    <div class="form-group">
                        <input type="search" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results" data-no-results-text="No results found"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
