# About

The Rice SNP-Seek Database and Web applications are developed using both open-source and commercial softwares.


## **Data Generation**
1.	BWA-MEM
2.	GATK
3.	VISTA

## **Storage**
1.	Oracle
2.	PostgreSQL
3.	HDF5

## **Web-application**
1.	Hibernate
2.	Spring
3.	ZK
4.	JBrowse
5.	BioJava
6.	jsphySVG

## **Development**
1)   MyEclipse
2)   Read the Docs
3)   GitHub
4)   Docker


## **Publications**
	
### **Primary publication**
- Mansueto, et al. **Rice SNP-seek database update: new SNPs, indels, and queries**.Nucl. Acids Res.(2017) 45 (D1): D1075-D1081.
doi: <a href="http://nar.oxfordjournals.org/content/early/2016/11/29/nar.gkw1135.abstract" target="_blank">10.1093/nar/gkw1135</a><br>
    *This paper describes new features and datasets added to SNP-Seek in 2015-2017 as well as software and database updates.*

### **Other publications**
- Mansueto, et al. **SNP-Seek II: A resource for allele mining and analysis of big genomic data in Oryza sativa**.Current Plant Biology(2016). 7-8:16-25
doi: <a href="http://doi.org/10.1016/j.cpb.2016.12.003" target="_blank">10.1016/j.cpb.2016.12.003</a><br>
  *This paper contains details on variant calling for 5 references, integration of additional genomic data, web interface, database schema, use cases, web services API.*<br>

- Alexandrov, et al. **SNP-Seek database of SNPs derived from 3000 rice genomes**.Nucl. Acids Res.(2015). 43(D1):D1023-D1027
doi: <a href="http://doi.org/10.1093/nar/gku1039" target="_blank">10.1093/nar/gku1039</a><br>
  *This is the first publication introducing SNP-Seek database, data generation for Nipponbare reference,and common use cases.*<br>

- Wang W, Mauleon R, Hu Z, Chebotarov D, Tai S, et al. **Genomic variation in 3,010 diverse accessions of Asian cultivated rice**. Nature.(2018). 557:43-49
doi: <a href="https://www.nature.com/articles/s41586-018-0063-9" target="_blank">10.1038/s41586-018-0063-9</a><br>
  *The main scientific paper for the 3K RGP.*