

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>User Login &mdash; SNP-Seek v3 Manual 1.0 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=b86133f3" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=f2a433a1"></script>
      <script src="_static/doctools.js?v=9bcbadda"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
      <script src="_static/js/external_links.js?v=c52a64b5"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            SNP-Seek v3 Manual
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Quick Start Guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="getting_started/index.html">Getting Started</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Frequently Asked Questions</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="faq/index.html">Sequence Download</a></li>
<li class="toctree-l1"><a class="reference internal" href="faq/index.html#haplotype-analysis">Haplotype Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="faq/index.html#reporting-bugs-and-communication-with-snp-seek-team">Reporting Bugs and Communication with SNP-Seek Team</a></li>
<li class="toctree-l1"><a class="reference internal" href="faq/index.html#other-questions">Other Questions</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">SNP-Seek v3 Manual</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">User Login</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index_backup.md.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <p><strong>SNPSeek v3 User Manual</strong></p>
<p><strong>QTL of Interest:
<a class="reference external" href="https://www.mdpi.com/2223-7747/10/8/1709"><u>PHS</u></a></strong></p>
<p><strong>Publication:</strong> Lee JS, Chebotarov D, McNally KL, Pede V, Setiyono TD,
Raquid R, Hyun WJ, Jeung JU, Kohli A, Mo Y. <em><strong>Novel Sources of
Pre-Harvest Sprouting Resistance for Japonica Rice Improvement</strong>. Plants
(Basel). 2021 Aug 19;10(8):1709. doi: 10.3390/plants10081709. PMID:
34451754; PMCID: PMC8401653.</em></p>
<section id="user-login">
<h1>User Login<a class="headerlink" href="#user-login" title="Link to this heading"></a></h1>
<p><img src="img/image1.png"
style="width:6.5in;height:3.4526in" /></p>
<ol class="arabic simple">
<li><p>Login using your SNP-seek credentials. There are three ways to do
this:</p>
<ol class="arabic simple">
<li><p>Username/email and Password</p></li>
<li><p>Sign in using your Google account</p></li>
<li><p>Sign in using <a class="reference external" href="http://cgiar.org"><u>cgiar.org</u></a> (if you are
part of this domain)</p></li>
</ol>
</li>
<li><p>Sign up if you do not have an account in SNPSeek yet.</p></li>
</ol>
</section>
<section id="search-genotypes">
<h1>Search Genotypes:<a class="headerlink" href="#search-genotypes" title="Link to this heading"></a></h1>
<blockquote>
<div><p>Note that our QTL/trait of interest is in this region:
<strong><u>Chr4:4662701-4670717</u></strong></p>
<p>You can begin by searching for SNPS that lie within this region using
the 3k rice dataset. There are 4 SNP Sets available:</p>
</div></blockquote>
<ul class="simple">
<li><p><strong><u>3K All</u>:</strong> 32 million full 3K RG SNPs Dataset</p></li>
</ul>
<blockquote>
<div><p>This SNP Set contains the full set of 32 million biallelic &amp;
multiallelic SNP.</p>
<p>Total SNPs: 32,064,217</p>
<p>Samples : 3024</p>
</div></blockquote>
<ul class="simple">
<li><p><strong><u>3kbase</u>:</strong> 18 million base SNP</p></li>
</ul>
<blockquote>
<div><p>The Base SNP set of ~18 million SNPs was created from the ~29 million
biallelic SNPs subset from the 32M full SNP set by removing SNPs with
excess of heterozygous calls.</p>
</div></blockquote>
<ul class="simple">
<li><p><strong><u>3K core</u></strong>: 404k CoreSNP dataset</p></li>
</ul>
<blockquote>
<div><p>The Core SNP set was obtained from the filtered SNP set by applying
two-step LD pruning procedure as follows:</p>
</div></blockquote>
<ol class="arabic simple">
<li><p>LD pruning with window size 10kb, step 1 SNP, R2 threshold 0.8</p></li>
<li><p>LD pruning with window size 50 SNPs, step 1 SNP, R2 threshold 0.8</p></li>
</ol>
<ul class="simple">
<li><p><strong><u>3k filtered</u>:</strong> 4.8million filtered SNP dataset</p></li>
</ul>
<blockquote>
<div><p>The filtered SNP set was obtained from the Base SNP set by applying
the following filtering criteria:</p>
</div></blockquote>
<ol class="arabic simple">
<li><p>alternative allele frequency at least 0.01</p></li>
<li><p>proportion of missing calls per SNP at most 0.2</p></li>
</ol>
<blockquote>
<div><p>You may use different SNP sets but for this hands-on, we will use the
<strong>“3Kbase” SNP set.</strong></p>
<p>In the Chromosome, put <strong>“Chr4”</strong>, in Start use <strong>“4662701”</strong> and end
<strong>“4670717”.</strong></p>
<p>Click <strong>“Search”</strong> to retrieve the positions within this region within
the 3KRG accessions.</p>
</div></blockquote>
<p><img src="img/image2.png"
style="width:1.91146in;height:3.02567in" /></p>
<p>Knowledge Check: How many SNPs were retrieved within the given region?</p>
</section>
<section id="query-result">
<h1>Query result<a class="headerlink" href="#query-result" title="Link to this heading"></a></h1>
<blockquote>
<div><p>After querying the genotypes, you will have a table which displays the
alleles for each 3KRG variety in a row, for all SNP positions within
the specified region or gene. By default, the varieties are sorted by
decreasing number of allele mismatches. The resulting table can be
filtered using any of the column values, and sorted based on any
column value by clicking the column header.</p>
</div></blockquote>
<p><img src="img/image3.png"
style="width:6.94683in;height:3.49479in" /></p>
<p>It will also report a graph of the allele frequencies where you can
inspect the major and minor alleles per subpopulation for all SNP
positions from the query.</p>
<p><img src="img/image4.png"
style="width:6.5in;height:2.59722in" /></p>
<p>In another panel, you can also view the region in a genome browser.</p>
<p><img src="img/image5.png"
style="width:6.5in;height:2.29167in" /></p>
</section>
<section id="snp-effects">
<h1>SNP Effects<a class="headerlink" href="#snp-effects" title="Link to this heading"></a></h1>
<blockquote>
<div><p>The Genotype query also returns a table containing the SNP effects
detected on MSU gene models. This is calculated using the SNPEff
program.</p>
</div></blockquote>
<p><img src="img/image6.png"
style="width:6.5in;height:3.49281in" /></p>
</section>
<section id="haplotype-view">
<h1>Haplotype View<a class="headerlink" href="#haplotype-view" title="Link to this heading"></a></h1>
<blockquote>
<div><p>Another view for the result of the genotype query is the Haplotype
View. This allows us to know whether our haplotypes make sense.</p>
</div></blockquote>
<p><img src="img/image7.png"
style="width:6.5in;height:3.008in" /></p>
<blockquote>
<div><p>You should also see major grouping of alleles which contributes to the
haplotype.</p>
</div></blockquote>
<p><img src="img/image8.png"
style="width:6.5in;height:3.12878in" /></p>
<p>In this hands-on, you should see <strong>3 major haplotype groups</strong>.</p>
<p><img src="img/image9.png"
style="width:7.0427in;height:1.96354in" /></p>
<p>Knowledge Check: How many varieties belong to each major haplotype
group?</p>
</section>
<section id="downloading-the-results-table">
<h1>Downloading the results table<a class="headerlink" href="#downloading-the-results-table" title="Link to this heading"></a></h1>
<p>Download the table in “Plink” format by clicking on the “Plink” button.</p>
<blockquote>
<div><p>This will give you two files in a zipped folder: a ped and map file
that by now you should be familiar with.</p>
<p>The first few lines of your downloaded .map file should look like
this:</p>
</div></blockquote>
<p><img src="img/image10.png"
style="width:2.72917in;height:2.69in" />…</p>
<p>Likewise, the first few lines of your .ped file should be like this:</p>
<p><img src="img/image11.png"
style="width:6.23958in;height:3.53125in" /></p>
</section>
<section id="add-positions-in-my-list">
<h1>Add positions in my list<a class="headerlink" href="#add-positions-in-my-list" title="Link to this heading"></a></h1>
<blockquote>
<div><p>From the map file, remove the middle columns such that only the 1st
column and the last columns remain (chr, position).</p>
<p>Go to “My lists”. Select “SNP” in List Type. Click on “Create” to make
a new list.</p>
<p>Paste the chr-position table from the map file.</p>
<p>You may tick the box on “Verify if SNP is in Variant set”. However, in
this hands-on, we will not do that since we know that these SNPs
already came as a result of the genotype query.</p>
<p><img src="img/image12.png"
style="width:3.08767in;height:5.14612in" /></p>
<p>You will be able to see this new list on the left side of the panel.</p>
<p><img src="img/image13.png"
style="width:2.79038in;height:3.70313in" />
<img src="img/image14.png"
style="width:2.34896in;height:3.6686in" /></p>
</div></blockquote>
</section>
<section id="annotate-snp-list">
<h1>Annotate SNP List<a class="headerlink" href="#annotate-snp-list" title="Link to this heading"></a></h1>
<blockquote>
<div><p>Next thing we want to do is to look for candidate genes within the QTL
region in Nipponbare, go to “Search”-&gt; “Gene Loci”.</p>
<p>Set “Japonica nipponbare” as reference genome.</p>
<p>Set “SNP Annotator (My SNP List)” in Search.</p>
<p>Set “All” to search all Gene models available.</p>
<p>Tick the box to “Include gene interactions (RiceNetv2, PRIN)” and
“Gene Ontology” annotations. You may choose to add more to gather more
annotations for this region.</p>
</div></blockquote>
<p><img src="img/image15.png"
style="width:3.59497in;height:4.54688in" /></p>
<p>The result should be a table which contains the annotations that can be
grouped by:</p>
<ol class="arabic simple">
<li><p>Marker/Position</p></li>
</ol>
<p><img src="img/image16.png"
style="width:6.5in;height:1.81944in" /></p>
<ol class="arabic simple" start="2">
<li><p>Gene Locus</p></li>
</ol>
<p><img src="img/image17.png"
style="width:6.5in;height:1.44444in" /></p>
<ol class="arabic simple" start="3">
<li><p>QTL</p></li>
</ol>
<blockquote>
<div><p><img src="img/image18.png"
style="width:6.5in;height:1.95833in" /></p>
</div></blockquote>
</section>
<section id="saving-the-gene-list">
<h1>Saving the Gene List<a class="headerlink" href="#saving-the-gene-list" title="Link to this heading"></a></h1>
<p>Save the list as “qDTF_genes”</p>
<blockquote>
<div><p>Go to “My Lists” and you should see 3 gene loci when you search under
“Locus”.</p>
</div></blockquote>
<p><img src="img/image19.png"
style="width:6.5in;height:1.98611in" /></p>
</section>
<section id="get-gene-set-network">
<h1>Get Gene set/network<a class="headerlink" href="#get-gene-set-network" title="Link to this heading"></a></h1>
<blockquote>
<div><p>Using the gene list we generated, we will now do a gene enrichment
analysis to know if there are other genes previously reported that are
associated with our QTL of interest.</p>
<p>To do this, go to “Search”-&gt; “Gene Loci.</p>
<p>Set as Reference genome: “Japonica Nipponbare”.</p>
<p>Set Search to: “Gene set/networks”.</p>
<p>Set Gene Models to: “All”.</p>
<p>Set the My Locus List to: “qDTF2_genes”.</p>
<p>Tick “Ricenetv2 direct neighbors” and “AgriGO”.</p>
<p>Click on “Gene set/enrichment”.</p>
<p><img src="img/image20.png"
style="width:6.5in;height:1.95833in" /></p>
<p>Unfortunately, this query will not return any result. Click the “Go”
button beside the Ricenet option and it will give us this message:</p>
<p><em>“No ROC analysis, because the valid query set size &lt; 4.”</em></p>
<p>In this case, our query size is very small. You can explore nearby
regions and see if there are existing gene networks reported by
increasing the upper and lower bound positions.</p>
</div></blockquote>
<p>To know more abouthow to use Rice SNPSeek, read the User’s manual in the
Help section:
<a class="reference external" href="https://snpseek.irri.org/_help.zul"><em><u>https://snpseek.irri.org/_help.zul</u></em></a></p>
</section>


           </div>
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2025, Breeding Innovations and Informatics, Rice Breeding Innovations, International Rice Research Institute.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>